src/Init/System/IO.lean

```diff
@@ -236,6 +242,34 @@ MonadIOProcess.wait c
 end Proc
 -/
 
+structure AccessRight :=
+(read write execution : Bool := false)
+
+def AccessRight.flags (acc : AccessRight) : UInt32 :=
+let r : UInt32 := if acc.read      then 0x4 else 0;
+let w : UInt32 := if acc.write     then 0x2 else 0;
+let x : UInt32 := if acc.execution then 0x1 else 0;
+r.lor $ w.lor x
+
+structure FileRight :=
+(user group other : AccessRight := { })
+
+def FileRight.flags (acc : FileRight) : UInt32 :=
+let u : UInt32 := acc.user.flags.shiftLeft 6;
+let g : UInt32 := acc.group.flags.shiftLeft 3;
+let o : UInt32 := acc.other.flags;
+u.lor $ g.lor o
+
+namespace Access
+
+@[extern "lean_chmod"]
+constant Prim.setAccessRights (filename : @& String) (mode : UInt32) : IO Unit :=
```

Comment:
Did you mean to put this inside the `Access` namespace?

Reply:
Would you not use any namespace for the access rights?

Reply:
It seems redundant given the two function names, and I also find it a little confusing to nest `Prim`.